# Python-proposal

## Brief description of the scripts

The first python and the second scripts were written in order to automatize the detection and gene annotation of somatic variants (SNVs) in whole-exome sequencing (WES) data from matched tumour-normal samples. The thrird python script aims to integrate and correlate gene expression levels (FPKM) of the genes in which SNVs have been detected, allowing the identification of high and low expressed variant/mutated genes. 

Specifically, the first pipeline uses the *Strelka* package for the detection of SNVs, and the second script performs gene annotation in the “passed” somatic variants list using the *SNPEff* package. An annotated VCF file will be generated from *SNPEff* and is going to be used as input to the second script. The second script takes .csv file containing the gene expression levels (FPKM) for the tumour sample and the SNPEff output file and generates a new file containing only the FPKM levels of the genes present in the SNPEff output (genes containing SNVs).

## Input Data

Genomic DNA was extracted matched tumour and normal DNA. Samples were sent to McGillUniversity and Génome Québec Innovation Centre for sequencing. Mouse whole-exome sequencing (WES) was performed in a Illumina HiSeq 4000 PE100instrument. After sequencing, paired-end reads for each sample were trimmed using *TrimGalore!* for remotion of Illumina adapters sequences. Reads (fastaq format) were quality checked using *FastQC* and were further aligned to the mouse reference genome (*mm10*) using *Bowtie*. After alignment BAM files were generated:

Sample (.bam) | Type | Size
---| --- | ---

857_Tail.bam | *normal sample* | size: 5.6GB
857_Thymus.bam | *tumour sample* | size: 6.1GB

Gene expression levels for the tumor sample 857_Thymus was generated by RNA-sequencing. Sequencing was performed by McGillUniversity and Génome Québec Innovation Centre using a Illumina HiSeq 4000 PE100instrument. Similarly with WES analysis, reads were trimmed using *TrimGalore!* for remotion of Illumina adapters sequences. Reads were aligned using *TopHat* and transcripts assembly and abundance was determined using *Cufflinks* package.

RNA-seq data: 857_gene_expression.csv

## Objectives

1. To determine the single nucleotide variants (SNVs) and small insertions and deletions (InDELs) in the tumour sample filtering out germline variants using *Strelka* package;
2. To functionally annotate the filtered genetic variants detected by *Strelka* with *SNPEff*  using the UCSC RefSeq genome (*mm10*) as database.
3. To retrieve the gene expression levels in FPKM only for the genes in which presented "passed" "SNVs detected by Strelka. 
